### ISSUE TITLE
Labels for silent detector

### ISSUE THREAD

**Issue by Toku11** [henryxrl/Listening-to-Sound-of-Silence-for-Speech-Denoising#5]
Hi! First of all thank your sharing your code and your paper(It has been a pleasure to read it)  I have gone through your code looking for the area were you create the labels for your audio without success, could you explain me briefly how are you generating your labels to train it ? I understand you have divided in segments of 1/30 seconds, but my question is more related to: you are setting output of that network in 100 which means you need to label 100 segments.. Am I correct? if you split a second by 30 you have 30 segments per seconds, then 2 seconds == 60 segments ie. [1,0,0,0,1,1,1,1,1,0.....,1] saying which segment is speech ==1 and which is silent ==0 ... hopefully I am understanding correctly... so how is that you match the 100 segments?

**Comment by Toku11** [comment]
In your preprocessing you are creating a "bit mask" which is supposed to be your labels for the silent interval detection however your bitmmask is always 1 because you created it like '1' * file_info[FIELDS[5]], how is that you use it to train

**Comment by aloha0920** [comment]
Hi, same issue here. Did you figure it out yet?

**Comment by henryxrl** [comment]
@Toku11 You are correct. Each 1/0 bit indicates a 1/30 time interval. And yes, 1 means speech, 0 means silence. Each piece of training data is indeed 2 seconds which means 60 of those 1/0 bits.

In the `preprocessor_audioonly.py` file, the reason why it's setting all 1's is that this generates data for inference job, NOT for training (so all 1's doesn't matter - you'll get the correct 1/0 sequence as a result). For training, you'll have to provide the ground truth correctly labeled 1/0 sequence. You can get creative and create your own. My way of doing so is to look at the power of the speech signal and label each 1/30 time interval 0 (silence) if the power is below a certain threshold, and 1 if above. As a reference, you can take a look at the 'get_bitstream_better' function (commented out) from the `util.py` file in the preprocessing directory.
